examples/tutorials/starting/observation_clustering.py

```diff
@@ -0,0 +1,195 @@
+# are observations which have a zenith angle less than the median value,
+# whilst the blue points are observations above the median.
+#
+# `obs_A` and `obs_B` are both `~gammapy.data.Observations` objects which
+# can be utilised in the usual way to show the various properties of the
+# observations i.e. see the :doc:`/tutorials/data/cta` tutorial.
+#
+
+
+######################################################################
+# Hierarchical clustering of observations
+# ---------------------------------------
+#
+# This method shows how to cluster observations based on their IRF quantities,
+# in this case those that have a similar edisp and psf. The
+# `gammapy.data.utils.get_irfs_features` is utilised to achieve this. The
+# observations are then clustered based on this criteria using
+# `gammapy.utils.cluster.hierarchical_clustering`. The idea here is to minimise
+# the variance of both edisp and psf within a specific group to limit the error
+# on the quanitity when they are stacked at the dataset level.
+#
+# In this example, the irf features are computed for the `edisp-res` and
+# `psf-radius` at 1 TeV. This is stored as a `astropy.table.table.Table`, as shown below.
+#
+
+source_position = SkyCoord.from_name("PKS 2155-304")
+names = ["edisp-res", "psf-radius"]
+features_irfs = get_irfs_features(
+    observations, energy_true="1 TeV", position=source_position, names=names
+)
+print(features_irfs)
+
+######################################################################
+# The `gammapy.utils.cluster.hierarchical_clustering` then clusters
+# this table into `t` groups with a corresponding label for each group.
+# In this case, we choose to cluster the observations into two groups.
+# We can print this table to show the corresponding label which has been
+# added to the previous `feature_irfs` table.
+#
+
+
+features = hierarchical_clustering(features_irfs, fcluster_kwargs={"t": 2})
+print(features)
+
+######################################################################
+# Finally, `observations.group_by_label` creates `t`
+# `~gammapy.data.Observation` objects by grouping the similar labels.
+#
+
+obs_clusters = observations.group_by_label(features["labels"])
+print(obs_clusters)
+
+
+mask_1 = features["labels"] == 1
+mask_2 = features["labels"] == 2
+fix, ax = plt.subplots(1, 1, figsize=(7, 5))
+ax.set_ylabel("edisp-res")
+ax.set_xlabel("psf-radius")
+ax.plot(
+    features[mask_1]["edisp-res"],
+    features[mask_1]["psf-radius"],
+    "o",
+    color="green",
+    alpha=0.4,
+    label="Group 1",
+)
+ax.plot(
+    features[mask_2]["edisp-res"],
+    features[mask_2]["psf-radius"],
+    "o",
+    color="magenta",
+    alpha=0.4,
+    label="Group 2",
+)
+ax.legend()
+
+
+######################################################################
+# The groups here are divided by the quality of the IRFs values `edisp-res`
+# and `psf-radius`. The green and magenta points indicate how the observations
+# are grouped.
```

Comment:
minor suggestion towards a more colorblind friendly approach: instead of referring to colors, you could rather use different markers or solid/empty circles (regardless the color). Same for the plot above muon eff vs zenith angle